src/cmake-tools.ts

```diff
@@ -463,9 +481,21 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
 
     this._statusMessage.set(localize('ready.status', 'Ready'));
 
+    this.extensionContext.subscriptions.push(vscode.workspace.onDidOpenTextDocument(async td => {
+      const str = td.uri.fsPath;
+      const sourceDirectory = await this.sourceDir;
+      if ((str.endsWith("CMakeLists.txt") || str.endsWith(".cmake")) && lightNormalizePath(str).startsWith(sourceDirectory)) {
```

Comment:
Does it matter if it's in the workspace?  We should probably just log it?

Reply:
Oh, I guess it doesn't. For some reason I assumed this from the start.